integration-cli/docker_cli_images_test.go

```diff
@@ -201,3 +201,102 @@ func (s *DockerSuite) TestImagesEnsureDanglingImageOnlyListedOnce(c *check.C) {
 	}
 
 }
+
+// See issue docker/docker#8926
+func (s *DockerSuite) TestImagesEnsureDanglingImagesExcludePullingImages(c *check.C) {
+	testRequires(c, Network)
+	defer deleteImages("hello-world")
+
+	cmd := exec.Command(dockerBinary, "images", "-q", "-f", "dangling=true")
+	initialOut, _, err := runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("listing images failed with errors: %s, %v", initialOut, err)
+	}
+	images := strings.Split(initialOut, "\n")
+	sort.Strings(images)
+	imagesStr := strings.Join(images, " ")
+
+	signalchan := make(chan int)
+	go func() {
+		defer close(signalchan)
+		pullCmd := exec.Command(dockerBinary, "pull", "hello-world")
+		if out, _, err := runCommandWithOutput(pullCmd); err != nil {
+			c.Fatalf("Failed to pull %v: error %v, output %q", "hello-world", err, out)
```

Comment:
`c.Fatal` won't work in a goroutine.
I'd recommend changing your chan to `make(chan error)` and send the error out.